src/bandersnatch/master.py

```diff
@@ -60,6 +60,19 @@ def get(self, path, required_serial, **kw):
                         required_serial, path, got_serial
                     )
                 )
+                # HACK: The following attempts to purge the cache of the page we
+                # just tried to fetch. This works around PyPI's caches sometimes
+                # returning a stale serial for a package. Ideally, this should
+                # be fixed on the PyPI side, at which point the following code
+                # should be removed.
+                logger.debug(f"Issuing a PURGE for {path} to clear the cache")
+                try:
+                    self.session.request("PURGE", path, timeout=self.timeout)
+                except requests.exceptions.HTTPError:
+                    logger.warning(
+                        "Got an error when attempting to clear the cache", exc_info=True
+                    )
+
                 raise StalePage(
                     "Expected PyPI serial {} for request {} but got {}".format(
```

Comment:
Can we add to the exception message that we’ve sent a PURGE (more to remind me / other devs in the future)

Reply:
Yeah but iirc the stalepage exception is just swallowed, so the message isn't seen.

Reply:
It will if you have `stop-on-error` set in the config file.

Reply:
Done.